### ISSUE TITLE
Management Form data missing or tampered bug

### ISSUE THREAD

**Issue by ezrajrice** [tethysplatform/tethys#592]
The following error occurs when trying to create a new (second) Tethys user account after successfully creating one. The username and password fields are unique on the second attempted account creation. After this error has occurred, I am unable to create any other user account until I restart Tethys.
OS: Ubuntu 18.04 LTS
```
 File "./django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "./django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "./django/contrib/admin/sites.py", line 223, in inner
    return view(request, *args, **kwargs)
  File "./django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "./django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "./django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "./django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "./django/contrib/auth/admin.py", line 99, in add_view
    return self._add_view(request, form_url, extra_context)
  File "./django/contrib/auth/admin.py", line 126, in _add_view
    return super().add_view(request, form_url, extra_context)
  File "./django/contrib/admin/options.py", line 1645, in add_view
    return self.changeform_view(request, None, form_url, extra_context)
  File "./django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "./django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "./django/contrib/admin/options.py", line 1529, in changeform_view
    return self._changeform_view(request, object_id, form_url, extra_context)
  File "./django/contrib/admin/options.py", line 1571, in _changeform_view
    if all_valid(formsets) and form_validated:
  File "./django/forms/formsets.py", line 448, in all_valid
    valid &= formset.is_valid()
  File "./django/forms/formsets.py", line 301, in is_valid
    self.errors
  File "./django/forms/formsets.py", line 281, in errors
    self.full_clean()
  File "./django/forms/formsets.py", line 322, in full_clean
    for i in range(0, self.total_form_count()):
  File "./django/forms/formsets.py", line 110, in total_form_count
    return min(self.management_form.cleaned_data[TOTAL_FORM_COUNT], self.absolute_max)
  File "./django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "./django/forms/formsets.py", line 92, in management_form
    code='missing_management_form',
django.core.exceptions.ValidationError: ['ManagementForm data is missing or has been tampered with']
```

**Comment by swainn** [comment]
@ezrajrice Thanks for reporting. Can you provide a little more information:

- What version of Tethys is this? Is it an upgraded instance or a fresh installation?
- Is this occurring in a production installation? Development installation? Both?
- How are you adding the new users? (i.e. programmatically, through the admin pages, through sign up forms)
- Do you have any additional Django apps installed?

**Comment by ezrajrice** [comment]
This is a production installation running version 3.0. It was a fresh install. I was adding them through the admin pages. There are no additional Django apps installed. It seems to be sporadic now because the error didn't occur today when I was adding new users. I don't know what triggers the error. I can't reproduce this in 3.1 on my local machine.

**Comment by speleophysics** [comment]
I am running into the same issue. It seems to happen both with and without debug mode turned on. I am running using the tethys core docker image. I updated to the 3.1.1 image and the problem is the same. It doesn't happen always with the second user. Sometimes I can add several. Sometimes deleting a user and then trying to add another seems to trigger the error. I'm also working through the admin interface.

**Comment by swainn** [comment]
I have encountered this issue in the past but it has been fairly infrequent. I haven't dug into what is causing it though. Could one of you provide the full Django error readout when you encounter this in debug mode next time? I believe it has a copy-paste version.

**Comment by speleophysics** [comment]
This is the error readout I get when I go to the copy-paste version, though it seems it may be mostly the same as above:

Environment:


Request Method: POST
Request URL: http://localhost/admin/auth/user/add/

Django Version: 2.2.14
Python Version: 3.7.8
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django_gravatar',
 'bootstrap3',
 'termsandconditions',
 'tethys_config',
 'tethys_apps',
 'tethys_gizmos',
 'tethys_services',
 'tethys_compute',
 'tethys_quotas',
 'social_django',
 'guardian',
 'session_security',
 'captcha',
 'snowpenguin.django.recaptcha2',
 'rest_framework',
 'rest_framework.authtoken',
 'analytical',
 'channels',
 'revproxy')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'tethys_portal.middleware.TethysSocialAuthExceptionMiddleware',
 'tethys_portal.middleware.TethysAppAccessMiddleware',
 'session_security.middleware.SessionSecurityMiddleware')



Traceback:

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/admin/options.py" in wrapper
  606.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapped_view
  142.                     response = view_func(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  44.         response = view_func(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/admin/sites.py" in inner
  223.             return view(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapper
  45.         return bound_method(*args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/views/decorators/debug.py" in sensitive_post_parameters_wrapper
  76.             return view(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapper
  45.         return bound_method(*args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapped_view
  142.                     response = view_func(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/auth/admin.py" in add_view
  99.             return self._add_view(request, form_url, extra_context)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/auth/admin.py" in _add_view
  126.         return super().add_view(request, form_url, extra_context)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/admin/options.py" in add_view
  1645.         return self.changeform_view(request, None, form_url, extra_context)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapper
  45.         return bound_method(*args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapped_view
  142.                     response = view_func(request, *args, **kwargs)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/admin/options.py" in changeform_view
  1529.             return self._changeform_view(request, object_id, form_url, extra_context)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/contrib/admin/options.py" in _changeform_view
  1571.             if all_valid(formsets) and form_validated:

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in all_valid
  448.         valid &= formset.is_valid()

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in is_valid
  301.         self.errors

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in errors
  281.             self.full_clean()

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in full_clean
  322.         for i in range(0, self.total_form_count()):

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in total_form_count
  110.             return min(self.management_form.cleaned_data[TOTAL_FORM_COUNT], self.absolute_max)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/utils/functional.py" in __get__
  80.         res = instance.__dict__[self.name] = self.func(instance)

File "/opt/conda/envs/tethys/lib/python3.7/site-packages/django/forms/formsets.py" in management_form
  92.                     code='missing_management_form',

Exception Type: ValidationError at /admin/auth/user/add/
Exception Value: ['ManagementForm data is missing or has been tampered with']

**Comment by speleophysics** [comment]
These are the portal settings in the debug output:

```

ABSOLUTE_URL_OVERRIDES | {}
-- | --
ACCEPT_TERMS_PATH | '/terms/accept/'
ADMINS | ()
ALLOWED_HOSTS | ['localhost', '127.0.0.1']
ANALYTICS_CONFIG | {}
ANALYTICS_CONFIGS | {}
ANONYMOUS_USER_ID | -1
APPEND_SLASH | True
ASGI_APPLICATION | 'tethys_portal.routing.application'
AUTHENTICATION_BACKENDS | ('django.contrib.auth.backends.ModelBackend',  'guardian.backends.ObjectPermissionBackend')
AUTH_PASSWORD_VALIDATORS | '********************'
AUTH_USER_MODEL | 'auth.User'
BASE_DIR | './tethys_portal'
BYPASS_TETHYS_HOME_PAGE | '********************'
CACHES | {'default': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}}
CACHE_MIDDLEWARE_ALIAS | 'default'
CACHE_MIDDLEWARE_KEY_PREFIX | '********************'
CACHE_MIDDLEWARE_SECONDS | 600
CAPTCHA_CONFIG | {'RECAPTCHA_PRIVATE_KEY': '********************',  'RECAPTCHA_PUBLIC_KEY': '********************'}
CHANNEL_LAYERS | {'default': {'BACKEND': None}}
CSRF_COOKIE_AGE | 31449600
CSRF_COOKIE_DOMAIN | None
CSRF_COOKIE_HTTPONLY | False
CSRF_COOKIE_NAME | 'csrftoken'
CSRF_COOKIE_PATH | '/'
CSRF_COOKIE_SAMESITE | 'Lax'
CSRF_COOKIE_SECURE | False
CSRF_FAILURE_VIEW | 'django.views.csrf.csrf_failure'
CSRF_HEADER_NAME | 'HTTP_X_CSRFTOKEN'
CSRF_TRUSTED_ORIGINS | []
CSRF_USE_SESSIONS | False
DATABASES | {'default': {'ATOMIC_REQUESTS': False,              'AUTOCOMMIT': True,              'CONN_MAX_AGE': 0,              'ENGINE': 'django.db.backends.postgresql_psycopg2',              'HOST': 'db',              'NAME': 'portal_db',              'OPTIONS': {},              'PASSWORD': '********************',              'PORT': 5432,              'TEST': {'CHARSET': None,                       'COLLATION': None,                       'MIRROR': None,                       'NAME': None},              'TIME_ZONE': None,              'USER': 'caver'}}
DATABASE_ROUTERS | []
DATA_UPLOAD_MAX_MEMORY_SIZE | 2621440
DATA_UPLOAD_MAX_NUMBER_FIELDS | 1000
DATETIME_FORMAT | 'N j, Y, P'
DATETIME_INPUT_FORMATS | ['%Y-%m-%d %H:%M:%S',  '%Y-%m-%d %H:%M:%S.%f',  '%Y-%m-%d %H:%M',  '%Y-%m-%d',  '%m/%d/%Y %H:%M:%S',  '%m/%d/%Y %H:%M:%S.%f',  '%m/%d/%Y %H:%M',  '%m/%d/%Y',  '%m/%d/%y %H:%M:%S',  '%m/%d/%y %H:%M:%S.%f',  '%m/%d/%y %H:%M',  '%m/%d/%y']
DATE_FORMAT | 'N j, Y'
DATE_INPUT_FORMATS | ['%Y-%m-%d',  '%m/%d/%Y',  '%m/%d/%y',  '%b %d %Y',  '%b %d, %Y',  '%d %b %Y',  '%d %b, %Y',  '%B %d %Y',  '%B %d, %Y',  '%d %B %Y',  '%d %B, %Y']
DEBUG | True
DEBUG_PROPAGATE_EXCEPTIONS | False
DECIMAL_SEPARATOR | '.'
DEFAULT_CHARSET | 'utf-8'
DEFAULT_CONTENT_TYPE | 'text/html'
DEFAULT_DB | {'ATOMIC_REQUESTS': False,  'AUTOCOMMIT': True,  'CONN_MAX_AGE': 0,  'ENGINE': 'django.db.backends.postgresql_psycopg2',  'HOST': 'db',  'NAME': 'portal_db',  'OPTIONS': {},  'PASSWORD': '********************',  'PORT': 5432,  'TEST': {'CHARSET': None, 'COLLATION': None, 'MIRROR': None, 'NAME': None},  'TIME_ZONE': None,  'USER': 'caver'}
DEFAULT_EXCEPTION_REPORTER_FILTER | 'django.views.debug.SafeExceptionReporterFilter'
DEFAULT_FILE_STORAGE | 'django.core.files.storage.FileSystemStorage'
DEFAULT_FROM_EMAIL | 'webmaster@localhost'
DEFAULT_INDEX_TABLESPACE | ''
DEFAULT_TABLESPACE | ''
DISALLOWED_USER_AGENTS | []
EMAIL_BACKEND | 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_CONFIG | {}
EMAIL_HOST | 'localhost'
EMAIL_HOST_PASSWORD | '********************'
EMAIL_HOST_USER | ''
EMAIL_PORT | 25
EMAIL_SSL_CERTFILE | None
EMAIL_SSL_KEYFILE | '********************'
EMAIL_SUBJECT_PREFIX | '[Django] '
EMAIL_TIMEOUT | None
EMAIL_USE_LOCALTIME | False
EMAIL_USE_SSL | False
EMAIL_USE_TLS | False
ENABLE_OPEN_PORTAL | False
ENABLE_OPEN_SIGNUP | False
ENABLE_RESTRICTED_APP_ACCESS | False
FILE_CHARSET | 'utf-8'
FILE_UPLOAD_DIRECTORY_PERMISSIONS | None
FILE_UPLOAD_HANDLERS | ['django.core.files.uploadhandler.MemoryFileUploadHandler',  'django.core.files.uploadhandler.TemporaryFileUploadHandler']
FILE_UPLOAD_MAX_MEMORY_SIZE | 2621440
FILE_UPLOAD_PERMISSIONS | None
FILE_UPLOAD_TEMP_DIR | None
FIRST_DAY_OF_WEEK | 0
FIXTURE_DIRS | []
FORCE_SCRIPT_NAME | None
FORMAT_MODULE_PATH | None
FORM_RENDERER | 'django.forms.renderers.DjangoTemplates'
GRAVATAR_DEFAULT_IMAGE | 'retro'
GRAVATAR_DEFAULT_RATING | 'g'
GRAVATAR_DEFAULT_SIZE | '80'
GRAVATAR_DFFAULT_SECURE | True
GRAVATAR_SECURE_URL | 'https://secure.gravatar.com/'
GRAVATAR_URL | 'http://www.gravatar.com/'
IGNORABLE_404_URLS | []
INSTALLED_APPS | ('django.contrib.admin',  'django.contrib.auth',  'django.contrib.contenttypes',  'django.contrib.sessions',  'django.contrib.messages',  'django.contrib.staticfiles',  'django_gravatar',  'bootstrap3',  'termsandconditions',  'tethys_config',  'tethys_apps',  'tethys_gizmos',  'tethys_services',  'tethys_compute',  'tethys_quotas',  'social_django',  'guardian',  'session_security',  'captcha',  'snowpenguin.django.recaptcha2',  'rest_framework',  'rest_framework.authtoken',  'analytical',  'channels',  'revproxy')
INTERNAL_IPS | []
LANGUAGES | [('af', 'Afrikaans'),  ('ar', 'Arabic'),  ('ast', 'Asturian'),  ('az', 'Azerbaijani'),  ('bg', 'Bulgarian'),  ('be', 'Belarusian'),  ('bn', 'Bengali'),  ('br', 'Breton'),  ('bs', 'Bosnian'),  ('ca', 'Catalan'),  ('cs', 'Czech'),  ('cy', 'Welsh'),  ('da', 'Danish'),  ('de', 'German'),  ('dsb', 'Lower Sorbian'),  ('el', 'Greek'),  ('en', 'English'),  ('en-au', 'Australian English'),  ('en-gb', 'British English'),  ('eo', 'Esperanto'),  ('es', 'Spanish'),  ('es-ar', 'Argentinian Spanish'),  ('es-co', 'Colombian Spanish'),  ('es-mx', 'Mexican Spanish'),  ('es-ni', 'Nicaraguan Spanish'),  ('es-ve', 'Venezuelan Spanish'),  ('et', 'Estonian'),  ('eu', 'Basque'),  ('fa', 'Persian'),  ('fi', 'Finnish'),  ('fr', 'French'),  ('fy', 'Frisian'),  ('ga', 'Irish'),  ('gd', 'Scottish Gaelic'),  ('gl', 'Galician'),  ('he', 'Hebrew'),  ('hi', 'Hindi'),  ('hr', 'Croatian'),  ('hsb', 'Upper Sorbian'),  ('hu', 'Hungarian'),  ('hy', 'Armenian'),  ('ia', 'Interlingua'),  ('id', 'Indonesian'),  ('io', 'Ido'),  ('is', 'Icelandic'),  ('it', 'Italian'),  ('ja', 'Japanese'),  ('ka', 'Georgian'),  ('kab', 'Kabyle'),  ('kk', 'Kazakh'),  ('km', 'Khmer'),  ('kn', 'Kannada'),  ('ko', 'Korean'),  ('lb', 'Luxembourgish'),  ('lt', 'Lithuanian'),  ('lv', 'Latvian'),  ('mk', 'Macedonian'),  ('ml', 'Malayalam'),  ('mn', 'Mongolian'),  ('mr', 'Marathi'),  ('my', 'Burmese'),  ('nb', 'Norwegian Bokmål'),  ('ne', 'Nepali'),  ('nl', 'Dutch'),  ('nn', 'Norwegian Nynorsk'),  ('os', 'Ossetic'),  ('pa', 'Punjabi'),  ('pl', 'Polish'),  ('pt', 'Portuguese'),  ('pt-br', 'Brazilian Portuguese'),  ('ro', 'Romanian'),  ('ru', 'Russian'),  ('sk', 'Slovak'),  ('sl', 'Slovenian'),  ('sq', 'Albanian'),  ('sr', 'Serbian'),  ('sr-latn', 'Serbian Latin'),  ('sv', 'Swedish'),  ('sw', 'Swahili'),  ('ta', 'Tamil'),  ('te', 'Telugu'),  ('th', 'Thai'),  ('tr', 'Turkish'),  ('tt', 'Tatar'),  ('udm', 'Udmurt'),  ('uk', 'Ukrainian'),  ('ur', 'Urdu'),  ('vi', 'Vietnamese'),  ('zh-hans', 'Simplified Chinese'),  ('zh-hant', 'Traditional Chinese')]
LANGUAGES_BIDI | ['he', 'ar', 'fa', 'ur']
LANGUAGE_CODE | 'en-us'
LANGUAGE_COOKIE_AGE | None
LANGUAGE_COOKIE_DOMAIN | None
LANGUAGE_COOKIE_NAME | 'django_language'
LANGUAGE_COOKIE_PATH | '/'
LOCALE_PATHS | []
LOGGERS | {'django': {'handlers': ['console_simple'], 'level': 'WARNING'},  'tethys': {'handlers': ['console_verbose'], 'level': 'INFO'},  'tethys.apps': {'handlers': ['console_verbose'], 'level': 'INFO'}}
LOGGING | {'disable_existing_loggers': False,  'formatters': {'simple': {'format': '%(levelname)s %(message)s'},                 'verbose': {'format': '%(levelname)s:%(name)s:%(message)s'}},  'handlers': {'console_simple': {'class': 'logging.StreamHandler',                                  'formatter': 'simple'},               'console_verbose': {'class': 'logging.StreamHandler',                                   'formatter': 'verbose'}},  'loggers': {'django': {'handlers': ['console_simple'], 'level': 'WARNING'},              'tethys': {'handlers': ['console_verbose'], 'level': 'INFO'},              'tethys.apps': {'handlers': ['console_verbose'], 'level': 'INFO'}},  'version': 1}
LOGGING_CONFIG | {}
LOGIN_REDIRECT_URL | '/accounts/profile/'
LOGIN_URL | '/accounts/login/'
LOGOUT_REDIRECT_URL | None
MANAGERS | []
MEDIA_ROOT | ''
MEDIA_URL | ''
MESSAGE_STORAGE | 'django.contrib.messages.storage.fallback.FallbackStorage'
MESSAGE_TAGS | {10: 'alert-danger',  20: 'alert-info',  25: 'alert-success',  30: 'alert-warning',  40: 'alert-danger'}
MIDDLEWARE | ('django.contrib.sessions.middleware.SessionMiddleware',  'django.middleware.common.CommonMiddleware',  'django.middleware.csrf.CsrfViewMiddleware',  'django.contrib.auth.middleware.AuthenticationMiddleware',  'django.contrib.messages.middleware.MessageMiddleware',  'django.middleware.clickjacking.XFrameOptionsMiddleware',  'tethys_portal.middleware.TethysSocialAuthExceptionMiddleware',  'tethys_portal.middleware.TethysAppAccessMiddleware',  'session_security.middleware.SessionSecurityMiddleware')
MIGRATION_MODULES | {}
MONTH_DAY_FORMAT | 'F j'
NUMBER_GROUPING | 0
OAUTH_CONFIG | {}
OAUTH_CONFIGS | {}
PASSWORD_HASHERS | '********************'
PASSWORD_RESET_TIMEOUT_DAYS | '********************'
PREPEND_WWW | False
RECAPTCHA_PRIVATE_KEY | '********************'
RECAPTCHA_PUBLIC_KEY | '********************'
RESOURCE_QUOTA_HANDLERS | ('tethys_quotas.handlers.workspace.WorkspaceQuotaHandler',)
REST_FRAMEWORK | {'DEFAULT_AUTHENTICATION_CLASSES': ('rest_framework.authentication.TokenAuthentication',),  'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAuthenticated',)}
ROOT_URLCONF | 'tethys_portal.urls'
SECRET_KEY | '********************'
SECURE_BROWSER_XSS_FILTER | False
SECURE_CONTENT_TYPE_NOSNIFF | False
SECURE_HSTS_INCLUDE_SUBDOMAINS | False
SECURE_HSTS_PRELOAD | False
SECURE_HSTS_SECONDS | 0
SECURE_PROXY_SSL_HEADER | None
SECURE_REDIRECT_EXEMPT | []
SECURE_SSL_HOST | None
SECURE_SSL_REDIRECT | False
SECURITY_EXPIRE_AFTER | 1800
SECURITY_WARN_AFTER | 1500
SERVER_EMAIL | 'root@localhost'
SESSION_CACHE_ALIAS | 'default'
SESSION_CONFIG | {'SECURITY_EXPIRE_AFTER': 1800, 'SECURITY_WARN_AFTER': 1500}
SESSION_COOKIE_AGE | 1209600
SESSION_COOKIE_DOMAIN | None
SESSION_COOKIE_HTTPONLY | True
SESSION_COOKIE_NAME | 'sessionid'
SESSION_COOKIE_PATH | '/'
SESSION_COOKIE_SAMESITE | 'Lax'
SESSION_COOKIE_SECURE | False
SESSION_ENGINE | 'django.contrib.sessions.backends.db'
SESSION_EXPIRE_AT_BROWSER_CLOSE | True
SESSION_FILE_PATH | None
SESSION_SAVE_EVERY_REQUEST | False
SESSION_SECURITY_EXPIRE_AFTER | 900
SESSION_SECURITY_WARN_AFTER | 840
SESSION_SERIALIZER | 'django.contrib.sessions.serializers.JSONSerializer'
SETTINGS_MODULE | 'tethys_portal.settings'
SHORT_DATETIME_FORMAT | 'm/d/Y P'
SHORT_DATE_FORMAT | 'm/d/Y'
SIGNING_BACKEND | 'django.core.signing.TimestampSigner'
SILENCED_SYSTEM_CHECKS | []
SOCIAL_AUTH_ADMIN_USER_SEARCH_FIELDS | ['username', 'first_name', 'email']
SOCIAL_AUTH_LOGIN_ERROR_URL | '/accounts/login/'
SOCIAL_AUTH_LOGIN_REDIRECT_URL | '/apps/'
SOCIAL_AUTH_SLUGIFY_USERNAMES | True
STATICFILES_DIRS | ('./tethys_portal/static',)
STATICFILES_FINDERS | ('django.contrib.staticfiles.finders.FileSystemFinder',  'django.contrib.staticfiles.finders.AppDirectoriesFinder',  'tethys_apps.static_finders.TethysStaticFinder')
STATICFILES_STORAGE | 'django.contrib.staticfiles.storage.StaticFilesStorage'
STATIC_ROOT | '/var/lib/tethys_persist/static'
STATIC_URL | '/static/'
TEMPLATES | [{'BACKEND': 'django.template.backends.django.DjangoTemplates',   'DIRS': ['./tethys_portal/templates'],   'OPTIONS': {'context_processors': ['django.contrib.auth.context_processors.auth',                                      'django.template.context_processors.debug',                                      'django.template.context_processors.i18n',                                      'django.template.context_processors.media',                                      'django.template.context_processors.static',                                      'django.template.context_processors.tz',                                      'django.template.context_processors.request',                                      'django.contrib.messages.context_processors.messages',                                      'social_django.context_processors.backends',                                      'social_django.context_processors.login_redirect',                                      'tethys_config.context_processors.tethys_global_settings_context',                                      'tethys_apps.context_processors.tethys_apps_context',                                      'tethys_gizmos.context_processors.tethys_gizmos_context'],               'debug': True,               'loaders': ['django.template.loaders.filesystem.Loader',                           'django.template.loaders.app_directories.Loader',                           'tethys_apps.template_loaders.TethysTemplateLoader']}}]
TERMS_BASE_TEMPLATE | 'page.html'
TERMS_EXCLUDE_URL_LIST | {'/'}
TERMS_EXCLUDE_URL_PREFIX_LIST | {'/accounts/', '/admin/', '/terms/', '/oauth2/', '/handoff/'}
TEST_NON_SERIALIZED_APPS | []
TEST_RUNNER | 'django.test.runner.DiscoverRunner'
TETHYS_HOME | '/usr/lib/tethys'
TETHYS_PORTAL_CONFIG | {}
TETHYS_WORKSPACES_ROOT | '/var/lib/tethys_persist/workspaces'
THOUSAND_SEPARATOR | ','
TIME_FORMAT | 'P'
TIME_INPUT_FORMATS | ['%H:%M:%S', '%H:%M:%S.%f', '%H:%M']
TIME_ZONE | 'UTC'
USE_I18N | True
USE_L10N | True
USE_THOUSAND_SEPARATOR | False
USE_TZ | True
USE_X_FORWARDED_HOST | False
USE_X_FORWARDED_PORT | False
WSGI_APPLICATION | None
X_FRAME_OPTIONS | 'SAMEORIGIN'
YEAR_MONTH_FORMAT | 'F Y'


```

**Comment by swainn** [comment]
I found this stack overflow on the topic: 

https://stackoverflow.com/questions/25331500/managementform-data-is-missing-or-has-been-tampered-with

Also, this discussion in the documentation:

https://docs.djangoproject.com/en/2.2/topics/forms/formsets/#understanding-the-managementform

The strange thing is that as far as I know, the /admin/auth/user/add/ view is provided by the built-in user auth Django app. We have customized the /admin/auth/user/edit/ form, but not the /admin/auth/user/add/ form as far as I know.

**Comment by speleophysics** [comment]
I was just clicking "Save," and then coming back to add another user from the user list. Exactly when the error happens for me seems a bit random, but, currently, it always eventually happens (i.e. at least by the time I've added 3-4 users).

**Comment by swainn** [comment]
@ezrajrice and @speleophysics I've submitted a PR that should fix this issue. Should be merged soon and released with Tethys Platform 3.2 at the end of the month.
